hledger-lib/Hledger/Query.hs

```diff
@@ -270,41 +278,109 @@ defaultprefix = "acct"
 
 -- | Parse a single query term as either a query or a query option,
 -- or return an error message if parsing fails.
-parseQueryTerm :: Day -> T.Text -> Either String (Either Query QueryOpt)
-parseQueryTerm _ (T.stripPrefix "inacctonly:" -> Just s) = Right $ Right $ QueryOptInAcctOnly s
-parseQueryTerm _ (T.stripPrefix "inacct:" -> Just s) = Right $ Right $ QueryOptInAcct s
+parseQueryTerm :: Day -> T.Text -> Either String (Query, [QueryOpt])
+parseQueryTerm _ (T.stripPrefix "inacctonly:" -> Just s) = Right (Any, [QueryOptInAcctOnly s])
+parseQueryTerm _ (T.stripPrefix "inacct:" -> Just s) = Right (Any, [QueryOptInAcct s])
```

Comment:
The diff looks a bit messy here so I will add a short explanation of what's changed in most of these lines.

I changed the type of `parseQueryTerm` to `(Query, [QueryOpt])` as "expr:" prefixed queries can contain multiple query options and queries that are combined to a single query. Hence it was no longer possible to return a single query or a single query option. I believe this plays reasonably nice with other components, as there were very little direct invocations of this function.

The only 'functionality' that is lost by this change as far as I saw is the very specific behaviour for `not:option`, where 'option' is some query option. This no longer respects that the "not:" prefix ignores query option queries. Instead, it always negates a query. But I believe it would be a bit silly to write a query that is just `not:option`, right?